resources/lib/tubecast/youtube/app.py

```diff
@@ -22,6 +22,9 @@
 
 import xbmc
 
+# Ugly hack to have type hints without actually having to import them
+if False:
```

Comment:
Yeah I really hate how this looks and I suspect you will too but the benefits of type hints surely outweigh the ugliness if you ask me.

Reply:
I don't understand how this is supposed to do anything. Can you clarify/explain?

Reply:
Your suspicion is entirely correct, this doesn't do anything *at runtime*. These type hints are meant to be read by editors to provide some type safety.
Kodi (Python 2 in general) doesn't have the typing module so to avoid an `ImportError` we can't import it at runtime.
Some editors (like PyCharm) are still able to pick them up properly though.

I've since started to hate this hacky import though. I'm pretty sure it wouldn't even work with something like Mypy.
You're also right about the type hints being more of a Python 3 feature. They added [limited support for Python 2](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code "PEP 484") (the type comments) but I guess I'll have to wait for Kodi Matrix to use them properly again.

I'm going to remove the import but would you allow me to keep the type comments in? Not only do they help IDEs that support it, they also help the reader in my opinion.

Reply:
Comments are fine, don't hurt and help understanding the code 👍